### PR TITLE
added new dependencies

### PR DESCRIPTION
**Pull request by Abinash-bit**
## PR Description

Fixes https://github.com/sunpy/sunkit-instruments/issues/134

## Changes Made:

1. astropy
2. packaging
3. scipy
4. numpy
5. h5py
6. matplotlib
7. pandas
Screenshot 2024-11-23 175301
Screenshot 2024-11-23 175332
Screenshot 2024-11-23 180558

### CODE REVIEW

**Comment by nabobalis** [comment]
Fixes https://github.com/sunpy/sunkit-instruments/issues/134

File: pyproject.toml [hunk 1]
```diff
@@ -18,6 +18,13 @@ authors = [
 dependencies = [
   "sunpy[map,net,timeseries,visualization]>=6.0.0",
   "xarray>=2023.12.0",
+  "astropy",
```
Comment (nabobalis) [inline pyproject.toml:21]: These will all need minimum version like xarray and sunoy do above.

**Comment by Abinash-bit** [comment]
> What is that check_pyproject.toml file doing?

That was a script i have written to check the lint syntax of .toml file .

File: pyproject.toml [hunk 2]
```diff
@@ -18,6 +18,13 @@ authors = [
...
+  "packaging",
```
Comment (nabobalis) [inline pyproject.toml:22]: Do we need this one? Does it not come with pip these days?

Comment (Abinash-bit) [reply pyproject.toml:22]: This one refers tp packaging right?

File: pyproject.toml [hunk 3]
```diff
@@ -18,6 +18,13 @@ authors = [
 dependencies = [
   "sunpy[map,net,timeseries,visualization]>=6.0.0",
   "xarray>=2023.12.0",
+  "astropy>=5.3.0",
+  "packaging>=23.0",
+  "scipy>=1.11.0",
+  "numpy>=1.24.0",
+  "h5py>=3.9.0",
+  "matplotlib>=3.7.0",
+  "pandas>=2.0.0",
```
Comment (nabobalis) [inline pyproject.toml:27]:
```suggestion
  "astropy>=5.3.0",
  "packaging>=23.0",
  # !=1.10.0 due to https://github.com/scipy/scipy/issues/17718
  "scipy>=1.9.0,!=1.10.0",
  "numpy>=1.23.5",
  "h5py>=3.7.0",
  "matplotlib>=3.6.0",
  "pandas>=1.4.0",
```

**Comment by nabobalis** [comment]
Looks like there is a warning that the oldestdeps is now raising: `RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility`. We just need to add this to the ignore list in the pytest.ini file.

**Comment by Abinash-bit** [comment]
There is an issue with codecov/project CI how can i proceed further?

**Comment by nabobalis** [comment]
Thanks @Abinash-bit

**Pull request merged**